Add support for ArrayAccess to __::has()

allejo opened the pull request:
Similar to the reasoning behind #41, `__::has()` should call `offsetExists()` for any objects that implement the ArrayAccess interface.

MonsieurV commented on src/__/collections/has.php:
```diff
@@ -5,7 +5,7 @@
  * Return true if $collection contains the requested $key.
  *
- * In constrast to isset(), __::has() returns true if the key exists but is null.
+ * In contrast to isset(), __::has() returns true if the key exists but is null.
```
Well spot!

MonsieurV commented on src/__/collections/has.php:
```diff
@@ -29,6 +29,12 @@ function has($collection, $path)
 //         $has = \__::isObject($collection) ? 'property_exists' : 'array_key_exists';
```
I think you can remove this commented code. (Code before the _numeric keys for objects_ issue was found, see comment below)

It certainly won't be used soon.

allejo replied:
Removed 👍

allejo commented:
> Specifically, ensure the path is parsed / splitted the same way

@MonsieurV what did you have in mind?

1. Change `__::has()` and `__::set()` to behave just like `__::get()` and use a for loop to go down each level?
1. Change `__::get()` to become a recursive call like `__::has()` and `__::set()`?

Or am I misunderstanding what you meant?

MonsieurV commented:
@allejo I was just thinking to create a common function to ensure the same separator and splitting method is used for nested indexes / path. So basically just a dumb factorization of `\__::split($path, '.')` (or just the token in a common constant).

Indeed we could go further by trying to factorize more of `__::has()`, `__::get()` and `__::set()`, but I wasn't pushing for that. (nor against it)

Do it if you feel it relevant, else the PR is now good for me!

allejo commented:
Moved it into a const, since at the moment, `__::split()` is used differently in `__::get()` than in `__::has()` and `__::set()` with the `$limit`.

`__::get()` has a TODO for making it recursive, so I can tackle that in a separate PR and see what can be shared.
